src/components/safe-apps/AppFrame/index.tsx

```diff
@@ -44,6 +59,36 @@ const AppFrame = ({ appUrl }: AppFrameProps): ReactElement => {
     setAppIsLoading(false)
   }, [appUrl, iframeRef, setAppIsLoading])
 
+  useEffect(() => {
+    const unsubscribe = txSubscribe(TxEvent.SAFE_APPS_REQUEST, async ({ txId, requestId }) => {
+      const currentRequestId = signMessageModalState.requestId || txModalState.requestId
+
+      if (txId && currentRequestId === requestId) {
+        const { detailedExecutionInfo } = await getTransactionDetails(chainId, txId)
+
+        if (isMultisigExecutionDetails(detailedExecutionInfo)) {
+          communicator?.send({ safeTxHash: detailedExecutionInfo.safeTxHash }, requestId)
+        }
+
+        txModalState.isOpen ? closeTxModal() : closeSignMessageModal()
+      }
+    })
+
+    return () => {
+      unsubscribe()
+    }
+  }, [chainId, closeTxModal, closeSignMessageModal, communicator, txModalState, signMessageModalState])
```

Comment:
Instead of passing `txModalState, signMessageModalState` you could just pass the id.
```
const requestId = txModalState.requestId || signMessageModalState.requestId
```

It would make it a little more clear why this hook depends on those two states.

Reply:
yes but even with that i have the `txModalState.isOpen` inside the effect so will need to create another effect for handling this

Reply:
Ah, I haven't noticed that one.